src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1465,6 +1467,117 @@ class Simplex_tree {
     }
   }
 
+  /** \brief Retrieve good values for extended persistence, and separate the 
+   * diagrams into the ordinary, relative, extended+ and extended- subdiagrams. 
+   * Need extend_filtration to be called first!
+   * @param[in] dgm Persistence diagram obtained after calling this->extend_filtration 
+   * and this->get_persistence.
+   * @return A vector of four persistence diagrams. The first one is Ordinary, the 
+   * second one is Relative, the third one is Extended+ and the fourth one is Extended-.
+   */
+  std::vector<std::vector<std::pair<int, std::pair<double, double>>>> compute_extended_persistence_subdiagrams(const std::vector<std::pair<int, std::pair<double, double>>>& dgm){
+    std::vector<std::vector<std::pair<int, std::pair<double, double>>>> new_dgm(4); 
+    double x, y;
+    for(unsigned int i = 0; i < dgm.size(); i++){ 
+      int h = dgm[i].first; 
+      double px = dgm[i].second.first; 
+      double py = dgm[i].second.second;
+      if(std::isinf(py))  continue;
+      else{
+        if ((px <= -1) & (py <= -1)){
+          x = minval_ + (maxval_-minval_)*(px + 2); 
+          y = minval_ + (maxval_-minval_)*(py + 2); 
+          new_dgm[0].push_back(std::make_pair(h, std::make_pair(x,y))); 
+        }
+        if ((px >=  1) & (py >=  1)){
+          x = minval_ - (maxval_-minval_)*(px - 2); 
+          y = minval_ - (maxval_-minval_)*(py - 2); 
+          new_dgm[1].push_back(std::make_pair(h, std::make_pair(x,y))); 
+        }
+        if ((px <= -1) & (py >=  1)){
+          x = minval_ + (maxval_-minval_)*(px + 2); 
+          y = minval_ - (maxval_-minval_)*(py - 2);
+          if (x <= y){
+            new_dgm[2].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+          else{  
+            new_dgm[3].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+        }
+      }
+    }
+    return new_dgm;
+  }
+
+  /** \brief Extend filtration for computing extended persistence. 
+   * This function only uses the filtration values at the 0-dimensional simplices, 
+   * and computes the extended persistence diagram induced by the lower-star filtration 
+   * computed with these values. Note that after calling this function, the filtration 
+   * values are actually modified. The function compute_extended_persistence_subdiagrams 
+   * retrieves the original values and separates the extended persistence diagram points 
+   * w.r.t. their types (Ord, Rel, Ext+, Ext-) and should always be called after 
+   * computing the persistent homology of the extended simplicial complex.
+   */
+  void extend_filtration() {
+
+    // Compute maximum and minimum of filtration values
+    int maxvert = -std::numeric_limits<int>::infinity();
+    std::vector<double> filt;
+    for (auto sh : this->complex_simplex_range()) {
```

Comment:
Indeed, thanks for the suggestion!

Reply:
Since we are actually inside the simplex tree, it would be possible to do:
```c++
for(auto sh=root_.members().begin(); sh!=root_.members().end(); ++sh)
```
(Simplex_handle happens to be a typedef for those iterators)
But I guess the skeleton range is good enough.